doc/users/next_whats_new/fig_draw_no_output.rst

```diff
@@ -0,0 +1,8 @@
+Figure now has draw_no_output method
+------------------------------------
+
+Rarely, the user will want to trigger a draw without making output to 
+either the screen or a file.  This is useful for determining the final 
+position of artists on the figure that require a draw like text.
+This could be accomplished via ``fig.canvas.draw()`` but that is 
+not user-facing, so a new method on `.Figure.draw_no_output` is provided.  
```

Comment:
This makes it sound like you could previously accomplish a *no output* draw with `fig.canvas.draw`, which afaict is only true if the user instantiated their own canvas and renderer, typically? 
```python
fig.canvas = backend_*.FigureCanvas*(fig)
renderer = backend_*.Renderer*(w, h, io.StringIO())
fig.draw(renderer)
```
or sth.

More importantly, `FigureCanvasBase.draw` is public, I guess I agree it's not a popular method probably, but I guess we shouldn't scare anyone reading this into thinking they weren't supposed to be using it.

```suggestion
None of the ways that currently exist to accomplish a draw without output 
were designed to be user-facing, so a new method,`.Figure.draw_no_output`, is 
provided.  
```

Reply:
`fig.canvas.draw()` always works, and is how constrained_layout etc work.   Its not meant to scare people from using it, its just saying why we are adding the convenience function - its buried on an attribute users don't usually consult, and hence don't know it exists.

Reply:
interesting I could have sworn that in the past I would try that and it would complain that there is no canvas at all, or that there is no renderer at all...


but you're the expert so if it's true and I'm totally off base ofc leave it as is

Reply:
> maybe something like finalize_artists/layout

This is not really finalizing anything though - the typical use-case is getting the position and size of artists, or filling the ticks, so that you can *change* the layout!  I think `draw` but `no_output` is exactly what is happening here.  I guess we could complain that `draw` is jargon, but I think its more discoverable than `finalize`.  Maybe there is something else?

Reply:
> interesting I could have sworn that in the past I would try that and it would complain that there is no canvas at all, or that there is no renderer at all...

I mean, I can imagine its possible to get into that state if you are monkeying around at the low levels.  But `Figure.__init__` calls `FigureCanvasBase(self)` and  that sets the canvas.

Reply:
In retrospect I agree finalize is a bad choice, especially since it has existing programming connotations. 

After trying to come up with a better name for some time, I'm now onboard with `draw_no_output` unless someone else has a better idea.

Reply:
Definitely open to suggestions, but maybe just because I know this is `fig.canvas.draw()` keeping draw in there seems appropriate.

We definitely have a tension in the library between simplifying things by removing simple wrappers, and adding simple wrappers so features are discoverable.  Personally, I think asking the user to understand there is a `canvas` to look for a `draw` method on is asking too much.

Reply:
> I mean, I can imagine its possible to get into that state if you are monkeying around at the low levels. But Figure.__init__ calls FigureCanvasBase(self) and that sets the canvas.

This is relatively recent (3.2 or 3.3), before `Figure` would `__init__` with canvas as `None`.